ultravox/inference/infer.py

```diff
@@ -152,6 +175,9 @@ def _dataproc(self, sample: datasets.VoiceSample):
         inputs = {k: v.to(self.model.device) for k, v in inputs.items()}
         if "audio_values" in inputs:
             inputs["audio_values"] = inputs["audio_values"].to(dtype=self.dtype)
+        if batch:
+            for key, val in inputs.items():
+                inputs[key] = val.squeeze(0)
         return inputs
```

Comment:
How does it compare to always assume `batch=True` or `batch=False`, i.e.., removing the `batch` parameter, and update the `infer()`, `infer_stream()`, `infer_batch()` methods accordingly? Logically, the output of `_dataproc` is processed by `_generate()`

Reply:
Yeah I think this will probably be cleaner, because the squeeze(0) is more tightly coupled with the input into collator as opposed to what data_proc is supposed to output.

Reply:
Although, this adds another loop through the already processed inputs. What do you think about the new change?